src/ripple/protocol/impl/STParsedJSON.cpp

```diff
@@ -542,18 +542,20 @@ static boost::optional<detail::STVar> parseLeaf (
 
                         std::string const strValue (account.asString ());
 
-                        if (value.size () == 40) // 160-bit hex account value
-                            uAccount.SetHex (strValue);
+                        // If we have what looks like a 160-bit hex account
+                        // value, try it and see if it's valid.
+                        bool const success = (value.size () == 40)
```

Comment:
True, however I feel that the salient point of the check is lost like that: namely that the `SetHex` call only makes sense if the value is exactly 40 characters long.

However, it occurs to me that there's a better option here. Update incoming.